locator/token_metadata.cc

```diff
@@ -59,7 +59,17 @@ class token_metadata_impl final {
     std::unordered_map<inet_address, inet_address> _replacing_endpoints;
 
     using ring_remapping = boost::icl::interval_map<token, std::unordered_set<inet_address>>;
-    std::unordered_map<sstring, ring_remapping> _pending_ranges_interval_map;
+    // For each keyspace, migration_info contains ranges of tokens and
+    // corresponding replicas to which writes or reads will be directed.
+    // This data structure is filled only during data migration between nodes
+    // when they are added or removed from the cluster.
+    // During normal operation, token mapping to nodes is
+    // implemented in the effective_replication_map.
```

Comment:
The comment suggests that the structure stores exactly the replicas to which we'll write:
```
     ... migration_info contains ranges of tokens and
    // corresponding replicas to which writes or reads will be directed.
```
But the truth is that `write_remapping` only contains the pending replicas, which are added to the natural replicas (obtained from effective_replication_map) during writes.

I still think we should not use the name "remapping" for writes because it's confusing, it suggests that we replace the usual replica set with something completely different, while we're summing both. `pending_replicas` or `pending_ranges` or `additional_replicas` are all appropriate.